geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/view/MergedDataStoreView.scala

```diff
@@ -19,14 +19,15 @@ import org.locationtech.geomesa.index.view.MergedQueryRunner.MergedStats
 import org.locationtech.geomesa.utils.geotools.{SchemaBuilder, SimpleFeatureTypes}
 import org.opengis.feature.`type`.{GeometryDescriptor, Name}
 import org.opengis.feature.simple.SimpleFeatureType
+import org.opengis.filter.Filter
 
 /**
   * Merged querying against multiple data stores
   *
   * @param stores delegate stores
   * @param namespace namespace
   */
-class MergedDataStoreView(val stores: Seq[DataStore], namespace: Option[String] = None)
+class MergedDataStoreView(val stores: Seq[(DataStore, Option[Filter])], namespace: Option[String] = None)
```

Comment:
Hmm... looks a little rough here.  Seems like one could have a filter per featureSource in the DataStore.  Since it covers the use, Imma ship it!

Reply:
you want to specify a filter per feature source? what would be the use case?

Reply:
I'm imagining a situation where several types are in the same DataStore.